x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java

```diff
@@ -374,7 +374,7 @@ static long computeDelay(int currentRetry, long maxRetryDelayInMillis) {
         return Math.min(backOffDelay, maxRetryDelayInMillis);
     }
 
-    private static boolean shouldRetry(Exception e) {
+    static boolean shouldRetry(Exception e) {
```

Comment:
Maybe `fetchGlobalCheckpoint()` should be done in `ShardFollowNodeTask` like we update mapping or fetch document from shard changes api then we don't need to make to method package protected and complete reuse the retry logic that exists in `ShardFollowNodeTask`?

Reply:
On the fence on this one. I understand your reasons but I feel that ShardFollowNodeTask is complicated enough so we shouldn't move things in there unless strictly needed. As far as I can tell, the amount of code stays the same.

Reply:
I see, I'm still undecided. Lets leave it this way.